netsim/modules/dhcp.py

```diff
@@ -329,6 +329,9 @@ def link_pre_transform(self, link: Box, topology: Box) -> None:
 
         # Copy link DHCP client status to interface in case the node does not have DHCP module enabled
         if link.get(f'dhcp.client.{af}',False) and intf.get(f'dhcp.client.{af}',None) is not False:
+          vlan_intf = intf.get('_vlan_mode',None)
+          if vlan_intf in ['bridge','irb']:         # No automatic DHCP clients on bridging/IRB interfaces
+            continue
```

Comment:
add a warning: Ignoring user provided dhcp.client.ipv[x] flag on interface

Reply:
I understand what you're saying, but there a whole saga behind the "dhcp.client" -- it is set everywhere (including the DHCP relay) and then cleaned up as needed. If we were to set it here, we'd have to modify the cleanup code as well, tightly coupling DHCP and VLAN module.

From a more realistic perspective: having a DHCP client on a VLAN interface on a VLAN-enabled LAN is less usual than not having it. Usually, the VLAN interface would be the DHCP relay.